consensus/state_sync/state_sync.go

```diff
@@ -0,0 +1,104 @@
+package state_sync
+
+import (
+	typesCons "github.com/pokt-network/pocket/consensus/types"
+	cryptoPocket "github.com/pokt-network/pocket/shared/crypto"
+	"github.com/pokt-network/pocket/shared/modules"
+)
+
+type StateSyncModule interface {
+	modules.Module
+
+	// -- Constructor Setter Functions --
+
+	// `HandleBlock` function contract:
+	// - Create a Utility Context
+	// - Block.ValidateBasic()
+	// - Consensus Module Replica Path
+	//   - Prepare Block:  utilityContext.SetProposalBlock(block)
+	//   - Apply Block:    utilityContext.ApplyBlock(block)
+	//   - Validate Block: utilityContext.AppHash == Block.AppHash
+	//   - Store Block:    consensusModule.CommitBlock()
+	SetHandleBlock(HandleBlock func(block *typesCons.Block) error)
+
+	// `GetPeerSyncMeta` function contract:
+	// - Retrieve a list of active peers with their metadata (identified and retrieved through P2P's `Churn Management`)
+	SetGetPeerMetadata(GetPeerSyncMeta func() (peers []PeerSyncMeta, err error))
+
+	// `NetworkSend` function contract:
+	// - sends data to an address via P2P network
+	SetNetworkSend(NetworkSend func(data []byte, address cryptoPocket.Address) error)
+
+	// -- Sync modes --
```

Comment:
Alternate design pattern for sync modes.

If you think of `mode` as an enum, we can do:

```go
StartMode(mode)
StopMode(mode)
```

This was we can (for example) turn off `ServerMode` if we need to based on some configs and a state the node enters.

Lmk what you think.

Reply:
It might be better to implement `modes` as `modules` so they have **Service Like** interfaces like `module.Start() and module.Stop()` 

WYT?

Reply:
Yes! We simply stop synching once it's synched!!!

Reply:
Perf - the interface implements `modules.Module`